File: C/Réseaux/client8.4.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
#include <arpa/inet.h>
#include <sys/socket.h>
#include <sys/types.h>
#include <fcntl.h>

#define PORT 12347
#define BUFFER 4096

int main() {
    int sockfd;
    struct sockaddr_in serv_addr;
    char buffer[BUFFER];

    // Créer socket
    sockfd = socket(AF_INET, SOCK_STREAM, 0);
    if(sockfd < 0){ perror("socket"); exit(1); }

    serv_addr.sin_family = AF_INET;
    serv_addr.sin_port = htons(PORT);
    inet_pton(AF_INET, "127.0.0.1", &serv_addr.sin_addr);

    // Connecter le client
    if(connect(sockfd, (struct sockaddr*)&serv_addr, sizeof(serv_addr)) < 0){
        perror("connect"); exit(1);
    }

    // Créer archive tar.gz du répertoire client
    system("tar zcvf /tmp/rep.tgz /chemin/vers/le/repertoire/client");

    // Ouvrir l'archive
    FILE *f = fopen("/tmp/rep.tgz", "rb");
    if(!f){ perror("fopen"); exit(1); }

    // Envoyer l'archive
    int n;
    while((n = fread(buffer, 1, BUFFER, f)) > 0){
        write(sockfd, buffer, n);
    }

    fclose(f);
    close(sockfd);
    printf("Archive envoyée au serveur !\n");
    return 0;
}

File: Python/Réseaux/client8.4.py
# client8.4.py
import socket
import os

HOST = '127.0.0.1'
PORT = 12347

# Chemin vers le répertoire client
rep_client = "/chemin/vers/le/repertoire/client"
archive = "/tmp/rep.tgz"

# Créer une archive tar.gz du répertoire
os.system(f"tar zcvf {archive} {rep_client}")

# Envoyer l'archive au serveur
with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
    s.connect((HOST, PORT))
    with open(archive, "rb") as f:
        data = f.read()
        s.sendall(data)

print("Archive envoyée !")